homeassistant/components/http.py

```diff
@@ -60,11 +66,19 @@ def setup(hass, config):
     development = str(conf.get(CONF_DEVELOPMENT, "")) == "1"
     ssl_certificate = conf.get(CONF_SSL_CERTIFICATE)
     ssl_key = conf.get(CONF_SSL_KEY)
+    allow_cors = conf.get(CONF_ALLOW_CORS, False)
+    cors_headers = conf.get(CONF_CORS_HEADERS, [HTTP_HEADER_ORIGIN,
```

Comment:
We should not allow configurable headers. It's our API and we know what is going on inside it.

Adding features for the sake of features is the worst thing to do. All of it will live in the code base _forever_. Every PR in the future will have to spin up an HTTP server to test functionality that no one might ever use.